File: 01-OOP-Refresher/src/go/rectangle.go
package main

import "fmt"

type Rectangle struct {
	Width  float64
	Height float64
}

func (r *Rectangle) Draw() {
	fmt.Printf("Drawing Rectangle (%.1f x %.1f)\n", r.Width, r.Height)
}

func (r *Rectangle) Area() float64 {
	return r.Width * r.Height
}

File: 01-OOP-Refresher/src/rust/rectangle.rs
use crate::drawable::Drawable;

pub struct Rectangle {
    pub width: f64,
    pub height: f64,
}

impl Drawable for Rectangle {
    fn draw(&self) {
        println!("Drawing Rectangle ({} x {})", self.width, self.height);
    }

    fn area(&self) -> f64 {
        self.width * self.height
    }
}
